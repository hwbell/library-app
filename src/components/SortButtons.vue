<template>

  <div id="sort-buttons">
    <p class="text">sort by: </p>

    <b-button class="sort" @click="sortReadingList('title')" >Title</b-button>
    <b-button class="sort" @click="sortReadingList('author')" >Author</b-button>
    <b-button class="sort" @click="sortReadingList('date')" >Newest</b-button>

  </div>

</template>

<script>
export default {
  name: "",
  props: {
  // 
  },
  data() {
    return {
      //
    };
  },
  methods: {
    sortReadingList(data) {
      console.log(`emitting sortReadingList with sortBy = ${data}`)
      this.$emit('sortReadingList', {
        sortBy: data
      })
    }
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style lang="scss" scoped>
// @import url("https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.7.0/animate.min.css");
#sort-buttons {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center; 
}
.text {
  margin: 6px;
}
.sort {
  margin-left: 4px;
  padding: 4px;
  border: none;
}

</style>