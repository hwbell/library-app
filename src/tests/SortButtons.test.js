// Import Vue and the component being tested
import { shallowMount, mount } from '@vue/test-utils'

import Vue from 'vue'
import SortButtons from '../components/SortButtons.vue';

describe('SortButtons', () => {

  it('renders correctly', () => {
    const wrapper = mount(SortButtons)
    expect(wrapper.element).toMatchSnapshot()
  })

  const wrapper = shallowMount(SortButtons)
  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('contains sort-buttons selector', () => {
    expect(wrapper.html()).toContain('id="sort-buttons"')
  })
  
  // test each of the buttons
  it('should emit sortReadingList with the correct data', () => {

    wrapper.find(".title-button").trigger('click')
    expect(wrapper.emitted().sortReadingList).toBeTruthy()
    expect(wrapper.emitted().sortReadingList[0]).toMatchObject([{"sortBy": "title"}])

    wrapper.find(".author-button").trigger('click')
    expect(wrapper.emitted().sortReadingList).toBeTruthy()
    expect(wrapper.emitted().sortReadingList[1]).toMatchObject([{"sortBy": "author"}])

    wrapper.find(".date-button").trigger('click')
    expect(wrapper.emitted().sortReadingList).toBeTruthy()
    expect(wrapper.emitted().sortReadingList[2]).toMatchObject([{"sortBy": "date"}])

  })

})

