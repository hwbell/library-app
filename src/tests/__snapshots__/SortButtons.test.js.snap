// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortButtons renders correctly 1`] = `
<div
  id="sort-buttons"
>
  <p
    class="text"
  >
    sort by: 
  </p>
   
  <b-button
    class="sort title-button"
  >
    Title
  </b-button>
   
  <b-button
    class="sort author-button"
  >
    Author
  </b-button>
   
  <b-button
    class="sort date-button"
  >
    Newer
  </b-button>
</div>
`;
